File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/datadriven/DataDrivenTests.cs
/// <filename>
///     DataDrivenTests.cs
/// </filename>
using datadriven;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;

[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class DataDrivenTests
{
    private datadriven.DataDrivenViewModel sut;

    private datadriven.DataDrivenViewModelTestEnvironment testEnvironment;

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
    public void SetUp()
    {
        this.testEnvironment = new datadriven.DataDrivenViewModelTestEnvironmentImpl();
        this.testEnvironment.Init();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void DataDrivenTable_given_Person_name_Name_age_42_when_then_()
    {
        System.Collections.Generic.List<string> NameExampleValues = new System.Collections.Generic.List<string> { };
        System.Collections.Generic.List<string> AgeExampleValues = new System.Collections.Generic.List<string> { };
        for (int exampleIndex = 1; exampleIndex <= 2; exampleIndex++)
        {
            this.given_Person_name_Name_age_42();
            this.BuildSut();
        }
    }

    protected virtual void BuildSut()
    {
        this.sut = this.testEnvironment.BuildSut();
    }

    private string Person_name_Name_age_42 = @"<Person name=""Name"" age=""42""></Person>";

    public virtual void given_Person_name_Name_age_42()
    {
        this.testEnvironment.SetXmlElementContext(this.Person_name_Name_age_42);
    }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/datadriven/DataDrivenTests.cpp
#include <gtest/gtest.h>
#include <memory>
#include "DataDrivenViewModelTestEnvironment.h"
#include "DataDrivenViewModel.h"
#include "DataDrivenViewModelTestEnvironmentImpl.h"
#include <vector>
#include <string>

namespace datadriven
{
  class DataDrivenTests : public testing::Test
  {
  public:
    std::shared_ptr<DataDrivenViewModel> sut;
    std::shared_ptr<DataDrivenViewModelTestEnvironment> testEnvironment;
    virtual void BuildSut();
    // Given Helper Definitions
    std::string Person_name_Name_age_42 = R"(<Person name="Name" age="42"></Person>)";
    virtual void given_Person_name_Name_age_42();
  protected:
    void SetUp() override;
  };
  void DataDrivenTests::SetUp()
  {
    this->testEnvironment = std::make_shared<DataDrivenViewModelTestEnvironmentImpl>();
    this->testEnvironment->Init();
  }
  /// Scenario: DataDrivenTable
  ///   given: Person_name_Name_age_42
  ///    when:
  ///    then:
  TEST_F(DataDrivenTests, DataDrivenTable_given_Person_name_Name_age_42_when_then_)
  {
    std::vector<std::string> NameExampleValues = { };
    std::vector<std::string> AgeExampleValues = { };
    for (int exampleIndex = 1; exampleIndex <= 2; exampleIndex++)
    {
      this->given_Person_name_Name_age_42();
      this->BuildSut();
    }
  }
  void DataDrivenTests::BuildSut()
  {
    this->sut = this->testEnvironment->BuildSut();
  }
  void DataDrivenTests::given_Person_name_Name_age_42()
  {
    this->testEnvironment->SetXmlElementContext(this->Person_name_Name_age_42);
  }
}
